### ISSUE TITLE
Vsix doesn't support VS 2019

### ISSUE THREAD

**Issue by Neme12** [dotnet/roslyn-analyzers#2034]
I'm getting this:

![image](https://user-images.githubusercontent.com/11444821/51662924-9ae8cb00-1fb5-11e9-933c-6d0364824f91.png)

**Comment by mavasani** [comment]
Seems like we need to publish a new VS2019 based VSIX...

**Comment by Neme12** [comment]
If that's the case then the current vsix should have supported version upper bound set to 16.0 so that we get a message right away saying that the extension doesn't support this version of visual studio as opposed to getting an error later during installation.

https://github.com/dotnet/roslyn-analyzers/blob/ba6242e405da3dba1f85d73d18c50bfb8eb7c846/src/Roslyn.Diagnostics.Analyzers/Setup/source.extension.vsixmanifest#L10

this should probably be `[15.0.27130.0,16.0)` as opposed to an open ended range?

**Comment by Neme12** [comment]
It would be really nice to have this now that VS 2019 is RTM.

**Comment by mavasani** [comment]
@Neme12 any reason you still using the VSIX instead of NuGet package? We are very likely not going to support this VSIX on VS2019 and recommend everyone to use the NuGet package instead.

**Comment by Neme12** [comment]
Why? :( I don't want to (and often can't) install a nuget package to every single project I'm working on or contributing to. I find many of the suggestions helpful - like making members static, but not everyone on my team will agree or want this to show up in their editor, because these analyzers aren't hard rules and aren't enforced.

**Comment by Neme12** [comment]
I want to make my code as pretty and consistent as possible but there are people that just don't care and don't want to be distracted. There's nothing I can do about that.

**Comment by Gerboa** [comment]
It doesn't seem like a **new feature**, especially when it's listed in the VS Marketplace as supporting VS2017, **VS2019**.

I also want to use the VSIX version rather than the NuGet package.

**Comment by Neme12** [comment]
What about just removing the arbitrary limitation for upper version of CoreEditor instead of creating a new VSIX?

**Comment by duncancoppedge** [comment]
It'd be great to have this on vs2019 or if not please update the [Visual Studio Marketplace page](https://marketplace.visualstudio.com/items?itemName=VisualStudioPlatformTeam.MicrosoftCodeAnalysis2017#overview) to reflect that it doesn't work. (see screenshot below)
Please note it appears ~32,000 other devs enjoy having the VSIX version.  I know I do 👍 

![image](https://user-images.githubusercontent.com/2111973/55642068-4ff0ee00-5784-11e9-97a8-77a1673fc986.png)

**Comment by Grasher134** [comment]
Any update on the issue? VS 2019 is released, but this vsix is still not supported

**Comment by mavasani** [comment]
Tagging @jinujoseph @vatsalyaagrawal - I believe @Neme12's [comment](https://github.com/dotnet/roslyn-analyzers/issues/2034#issuecomment-479584963) matches the reason why VSIX based support was added in first place, and it seems sense to add support for the VSIX in VS2019, given its popularity.

**Comment by mavasani** [comment]
I have a PR out to fix the VSIX: https://github.com/dotnet/roslyn-analyzers/pull/2349. Once it is merged, we will need to publish the VSIX and update the documentation. I'll keep using this issue to track all the work.

**Comment by mavasani** [comment]
Re-opening as the VSIX is not yet published.

**Comment by Korporal** [comment]
> @Neme12 any reason you still using the VSIX instead of NuGet package? We are very likely not going to support this VSIX on VS2019 and recommend everyone to use the NuGet package instead.

@mavasani - The only reason he needs is that VS 2019 provides this, offers this as an option - how are we to know that VS 2019 is offering us options that you intend to remove support for??

**Comment by sinand99** [comment]
Waiting for 2 weeks for a simple extension update :(  Is it so hard to update vsixmanifest and reupload ?

**Comment by mavasani** [comment]
@sinand99 - Sorry, working through the legal and publisher credentials aspects before pushing out the VSIX. I hope to publish it soon. Also, note that VSIX is still not our primary recommended means to get these analyzers. We strongly recommend using the FxCop analyzers NuGet package instead.

**Comment by Grasher134** [comment]
People update their VSIX packages with every VS version release. I don't see any problem with you doing the same. 

Having FxCop not in Nuget packages is a huge benefit, as it allows individual developer to benefit from it and improve, without interrupting team's work process.
